lib/ecto/query/builder.ex

```diff
@@ -261,9 +261,15 @@ defmodule Ecto.Query.Builder do
     {expr, params_acc}
   end
 
+  def escape({:count, _, []}, type, params_acc, vars, env) do
+    expr = {:{}, [], [:count, [], [:*]]}
```

Comment:
This should expand to `{:{}, [], [:count, [], []]}`.

Reply:
The idea here is that `count(:*)` is a behaviour that is specific to SQL databases. We should only work with `count()` and the database can convert that to `count(:*)` if they find necessary.